### PR TITLE
fix: Do not require git user being set when updating project upstream credentials

### LINKED ISSUE #7496: Git set upstream without `Git user` does not work
**Environment**

* Keptn Version (`keptn version`): `0.14.1`

**Affected Component**

* [x] Bridge

**Describe the bug**

* When trying to set an upstream without `Git user` (since it is not required anymore), the **Set Git upstream** button does not work. Please see the network trace that stays empty: 
image

**To Reproduce**

* Create a project without upstream repo
* Follow the link: 
image
* Enter `Git URL` and `Git Token`
* Click **Set Git upstream** button

**Expected behavior**

* Since the `Git user` is not mandatory anymore, I would expect that setting upstream without user works. 

**Current behavior**

When providing the `Git user`, the operation works: 
image

### PR DESCRIPTION
**Pull request by bacherfl**
Closes #7496 

A couple of adjustments had to be made in the shipyard-controller, as well as the configuration service in order to support setting the upstream without a user.

### CODE REVIEW

File: configuration-service/common/git.go [hunk 1]
```diff
@@ -519,32 +525,8 @@ func ServiceExists(project string, stage string, service string, disableUpstream
 
 // GetCredentials returns the git upstream credentials for a given project (stored as a secret), if available
 func GetCredentials(project string) (*common_models.GitCredentials, error) {
```
Comment (odubajDT) [inline configuration-service/common/git.go:527]: We probably do not have a reason to have this function, as we can use `credentialsReader.GetCredentials(project)`, but as it is configuration-service, there is probably no reason to refactor it and spent time on improving the code

File: configuration-service/common/git_test.go [hunk 2]
```diff
@@ -786,3 +795,55 @@ func TestGit_CloneRepo(t *testing.T) {
...
+		{
+			name: "malformed credentials",
+			args: args{
+				data: []byte(`invalid`),
+			},
+			want:    nil,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
```
Comment (odubajDT) [inline configuration-service/common/git_test.go:838]: please add test for input: `{"token":"token","remoteURI":"http://remote-url.git"}`

Comment (bacherfl) [reply configuration-service/common/git_test.go:849]: done

**Review by odubajDT — APPROVED** [review]
LGTM

**Pull request merged**